[docs] .NET in Gitpod (#494)
gitpod-io/website: src/docs/dotnet-in-gitpod.md

|[coolstore-microservices](https://github.com/vietnam-devs/coolstore-microservices)|A Kubernetes-based microservices application on service mesh written in C#|[![Open in Gitpod](https://gitpod.io/button/open-in-gitpod.svg)](https://gitpod.io/#https://github.com/vietnam-devs/coolstore-microservices)|
|[uno.quickstart](https://github.com/unoplatform/uno.quickstart)|An Uno "Hello world!" project using Windows UWP, iOS, Android and WebAssembly|[![Open in Gitpod](https://gitpod.io/button/open-in-gitpod.svg)](https://gitpod.io/#https://github.com/unoplatform/uno.quickstart)|
|[Fable](https://github.com/fable-compiler/Fable)|F# to JavaScript Compiler written in F#|[![Open in Gitpod](https://gitpod.io/button/open-in-gitpod.svg)](https://gitpod.io/#https://github.com/fable-compiler/Fable)|

</div>

## Installing the .NET Tools

To get the all the .NET tools that you need pre-installed in your Gitpod workspaces, use the following base image in your [.gitpod.yml](https://www.gitpod.io/docs/config-gitpod-file/):
```YAML
image: gitpod/workspace-dotnet
```

If your project requires a GUI or Virtual Desktop, please use this image instead:
```YAML
image: gitpod/workspace-dotnet-vnc
```

> This will give you a full GUI Virtual Desktop for your .NET application!

Reviewer: ```suggestion
This will give you a full GUI Virtual Desktop for your .NET application! The Virtual Desktop will be running on port `6080`.
```

## Compiling C# Code in Gitpod

To compile your C# Code in Gitpod run `mono-csc <FILE>`  where `<FILE>` is the path to the C# program you want to compile

## Compiling F# Code in Gitpod

> To compile your F# Code in Gitpod run `fsharpc <FILE>`  where `<FILE>` is the path to the F# program you want to compile

Reviewer: I feel like this would be less repetitive as a single section.

```suggestion
## Compiling .NET Code in Gitpod

To compile your **C#** Code in Gitpod, run `mono-csc <FILE>`,  where `<FILE>` is the path to the C# program you want to compile.

To compile your **F#** Code in Gitpod, run `fsharpc <FILE>`,  where `<FILE>` is the path to the F# program you want to compile.
```

# .NET in Gitpod

> ## Examples

Reviewer: Some guides have "Examples", some have "Examples Repositories".

I personally prefer the latter, but I also think it would be good to keep all the guides consistent to some extent.

Author: done!

> Before we begin here are some example repositories to give you an idea of what to expect!

Reviewer: Some guides seem to have:

> Here are some example <X> projects that are automated with Gitpod:

which works well. But it's very inconsistent (some guides don't even have a sentence there).

> Before we begin here are some example repositories to give you an idea of what to expect!

Author: Done!

<div class="table-container">

| Repository | Description | Try it |
|------|----------------|-----------|
|[uno](https://github.com/unoplatform/uno)|Build Mobile, Desktop and WebAssembly apps with C# and XAML|[![Open in Gitpod](https://gitpod.io/button/open-in-gitpod.svg)](https://gitpod.io/#https://github.com/unoplatform/uno)
|[coolstore-microservices](https://github.com/vietnam-devs/coolstore-microservices)|A Kubernetes-based microservices application on service mesh written in C#|[![Open in Gitpod](https://gitpod.io/button/open-in-gitpod.svg)](https://gitpod.io/#https://github.com/vietnam-devs/coolstore-microservices)|
|[uno.quickstart](https://github.com/unoplatform/uno.quickstart)|An Uno "Hello world!" project using Windows UWP, iOS, Android and WebAssembly|[![Open in Gitpod](https://gitpod.io/button/open-in-gitpod.svg)](https://gitpod.io/#https://github.com/unoplatform/uno.quickstart)|
|[Fable](https://github.com/fable-compiler/Fable)|F# to JavaScript Compiler written in F#|[![Open in Gitpod](https://gitpod.io/button/open-in-gitpod.svg)](https://gitpod.io/#https://github.com/fable-compiler/Fable)|

</div>

## Installing the .NET Tools

> To get the all the .NET you pre-installed in your Gitpod workspaces, use the following base image in your [.gitpod.yml](https://www.gitpod.io/docs/config-gitpod-file/):

Reviewer: This sentence doesn't make sense. Maybe you meant "tools" instead of "you"?

Author: Done!

# .NET in Gitpod

## Examples

Before we begin here are some example repositories to give you an idea of what to expect!

<div class="table-container">

| Repository | Description | Try it |
|------|----------------|-----------|
|[uno](https://github.com/unoplatform/uno)|Build Mobile, Desktop and WebAssembly apps with C# and XAML|[![Open in Gitpod](https://gitpod.io/button/open-in-gitpod.svg)](https://gitpod.io/#https://github.com/unoplatform/uno)
|[coolstore-microservices](https://github.com/vietnam-devs/coolstore-microservices)|A Kubernetes-based microservices application on service mesh|[![Open in Gitpod](https://gitpod.io/button/open-in-gitpod.svg)](https://gitpod.io/#https://github.com/vietnam-devs/coolstore-microservices)|
|[uno.quickstart](https://github.com/unoplatform/uno.quickstart)|An Uno "Hello world!" project using Windows UWP, iOS, Android and WebAssembly|[![Open in Gitpod](https://gitpod.io/button/open-in-gitpod.svg)](https://gitpod.io/#https://github.com/unoplatform/uno.quickstart)|

> |[Fable](https://github.com/fable-compiler/Fable)|F# to JavaScript Compiler|[![Open in Gitpod](https://gitpod.io/button/open-in-gitpod.svg)](https://gitpod.io/#https://github.com/fable-compiler/Fable)|

Reviewer: Since this guide is about multiple languages (C#, F#, Visual Basic) I think it's important to specify what some examples are written in.

```suggestion
|[coolstore-microservices](https://github.com/vietnam-devs/coolstore-microservices)|A Kubernetes-based microservices application on service mesh written in C#|[![Open in Gitpod](https://gitpod.io/button/open-in-gitpod.svg)](https://gitpod.io/#https://github.com/vietnam-devs/coolstore-microservices)|
|[uno.quickstart](https://github.com/unoplatform/uno.quickstart)|An Uno "Hello world!" project using Windows UWP, iOS, Android and WebAssembly|[![Open in Gitpod](https://gitpod.io/button/open-in-gitpod.svg)](https://gitpod.io/#https://github.com/unoplatform/uno.quickstart)|
|[Fable](https://github.com/fable-compiler/Fable)|F# to JavaScript Compiler written in F#|[![Open in Gitpod](https://gitpod.io/button/open-in-gitpod.svg)](https://gitpod.io/#https://github.com/fable-compiler/Fable)|
```

</div>

## Installing the .NET CLI

> To install the .NET CLI in your Gitpod workspace use the following base image in your [.gitpod.Dockerfile](https://www.gitpod.io/docs/config-docker/)

Reviewer: Phrasing suggestion (one is not actually "installing" something when downloading a Docker image, I think):

```suggestion
To get the .NET CLI pre-installed in your Gitpod workspaces, use the following base image in your [.gitpod.Dockerfile](https://www.gitpod.io/docs/config-docker/):
```

```Dockerfile
FROM gitpod/workspace-dotnet
```

> If your project requires a GUI or a desktop please use

Reviewer: ```suggestion
If your project requires a GUI or Virtual Desktop, please use this image instead:
```

```Dockerfile
FROM gitpod/workspace-dotnet-vnc
```

> This will give you a full GUI Desktop for your .NET application!

Reviewer: ```suggestion
This will give you a full GUI Virtual Desktop for your .NET application!
```

## <p>Installing F#</p>

> To compile your F# Program add the following to your [.gitpod.Dockerfile](https://www.gitpod.io/docs/config-docker/)

Reviewer: ```suggestion
To compile F# programs, add the following to your [.gitpod.Dockerfile](https://www.gitpod.io/docs/config-docker/):
```

```Dockerfile
RUN sudo apt-get update \
    && sudo apt-get install -y \

>         fsharp

Reviewer: ```suggestion
RUN sudo apt-get update && \
    sudo apt-get install -y fsharp && \
    sudo rm -rf /var/lib/apt/lists/*
```

```

> Then to compile the F# program run `fsharpc <FILE>` where `<FILE>` is the path to the F# program you want to compile

Reviewer: ```suggestion
Then to compile the F# program run `fsharpc <FILE>` where `<FILE>` is the path to the F# program you want to compile.
```

## Running .EXE files in Gitpod

> To run your code after it has been compiled add the following to your [.gitpod.Dockerfile](https://www.gitpod.io/docs/config-docker/)

Reviewer: ```suggestion
To be able to run your code after it has been compiled, add the following to your [.gitpod.Dockerfile](https://www.gitpod.io/docs/config-docker/):
```

```Dockerfile
RUN sudo apt-get update \
    && sudo apt-get install -y \

>         mono-complete

Reviewer: ```suggestion
RUN sudo apt-get update && \
    sudo apt-get install -y mono-complete && \
    sudo rm -rf /var/lib/apt/lists/*
```

```

> Then to run the program run `mono <FILE>` where `<FILE>` is the path to the .EXE file you want to run

Reviewer: ```suggestion
Then run `mono <FILE>` where `<FILE>` is the path to the .EXE file you want to run.
```

## Compiling C# Code in Gitpod

> To compile your C# Code in Gitpod you must (if you haven't already) install mono-complete see the snippet above, then to compile it run `mono-csc <FILE>`  where `<FILE>` is the path to the C# program you want to compile

Reviewer: This section seems pretty essential. Please move it up, above `Installing F#`.

Author: Done!!

> ## VSCode Extensions

Reviewer: Maybe a more specific title can be helpful (to tell why exactly these extensions are listed here)

```suggestion
## Useful VSCode Extensions
```

### <p>The C# Extension</p>

![C# Demo Extension](images/CSharpDemo.png)

> This extension will allow for code completion, snippets, autoformatting, peek definiton, refactoring, and hover documentation! To add this extension to your repository add the following to your [.gitpod.yml](https://www.gitpod.io/docs/config-gitpod-file/):

Reviewer: ```suggestion
This extension brings code completion, snippets, auto-formatting, peek definiton, refactoring, and hover documentation for C#! To install this extension for your repository, add the following to your [.gitpod.yml](https://www.gitpod.io/docs/config-gitpod-file/) file:
```

```yaml
vscode:
  extensions:
    - ms-vscode.csharp@1.21.12:kw8SkO8+aVTSFug281WfQQ==
```

### Ionide-fsharp

![Ionide-fsharp](images/Iondine_Example.png)

> Ionide-fsharp will add the following

Reviewer: ```suggestion
Ionide-fsharp gives F# developers the following features:
```

* A Widget for your F# needs 
* Auto completion
* Shows errors inline
* Quickfix
* Goto definiton
* Peek definiton

> To add this extension to your repository add the following to your [.gitpod.yml](https://www.gitpod.io/docs/config-gitpod-file/):

Reviewer: ```suggestion
To add this extension to your repository, add the following to your [.gitpod.yml](https://www.gitpod.io/docs/config-gitpod-file/):
```
